job/replay_syncer.go

```diff
@@ -0,0 +1,181 @@
+package job
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/odpf/optimus/core/logger"
+	"github.com/odpf/optimus/core/tree"
+	"github.com/odpf/optimus/models"
+	"github.com/odpf/optimus/store"
+	"github.com/pkg/errors"
+)
+
+var (
+	ReplayStatusToSynced = []string{models.ReplayStatusReplayed, models.ReplayStatusInProgress, models.ReplayStatusAccepted}
+	ReplayMessageSuccess = "all instances for this replay are successfully run"
+	ReplayMessageFailed  = "instance run failure found"
+)
+
+type ReplaySyncer interface {
+	Sync(context.Context, time.Duration) error
+}
+
+type Syncer struct {
+	replaySpecFactory         ReplaySpecRepoFactory
+	scheduler                 models.SchedulerUnit
+	dependencyResolver        DependencyResolver
+	projectJobSpecRepoFactory ProjectJobSpecRepoFactory
+	assetCompiler             AssetCompiler
+	Now                       func() time.Time
+	projects                  []models.ProjectSpec
+}
+
+func NewReplaySyncer(replaySpecFactory ReplaySpecRepoFactory, scheduler models.SchedulerUnit, dependencyResolver DependencyResolver, projectJobSpecRepoFactory ProjectJobSpecRepoFactory, assetCompiler AssetCompiler, projects []models.ProjectSpec) *Syncer {
+	return &Syncer{replaySpecFactory: replaySpecFactory, scheduler: scheduler, dependencyResolver: dependencyResolver, projectJobSpecRepoFactory: projectJobSpecRepoFactory, assetCompiler: assetCompiler, Now: time.Now, projects: projects}
```

Comment:
time should be injected in the constructor and with UTC

Reply:
do this mean 'Now' value should also passed to the constructor, not being initialized in the constructor?

Reply:
done by Kush